src/Tools/GenerateRulesMissingDocumentation/Program.cs

```diff
@@ -0,0 +1,173 @@
+string binDirectory = args[2];
+string configuration = args[3];
+
+var directory = Directory.CreateDirectory(analyzerDocumentationFileDir);
+var fileWithPath = Path.Combine(directory.FullName, "RulesMissingDocumentation.md");
+
+var builder = new StringBuilder();
+builder.Append(@"# Rules without documentation
+
+Rule ID | Missing Help Link | Title |
+--------|-------------------|-------|
+");
+
+var actualContent = Array.Empty<string>();
+if (validateOnly)
+{
+    actualContent = File.ReadAllLines(fileWithPath);
+}
+
+var allRulesById = GetAllRulesById(binDirectory, configuration);
+
+foreach (var ruleById in allRulesById)
+{
+    string ruleId = ruleById.Key;
+    DiagnosticDescriptor descriptor = ruleById.Value;
+
+    var helpLinkUri = descriptor.HelpLinkUri;
+    if (!string.IsNullOrWhiteSpace(helpLinkUri) &&
+        await checkHelpLinkAsync(helpLinkUri).ConfigureAwait(false))
+    {
+        // Rule with valid documentation link
+        continue;
+    }
+
+    // The angle brackets around helpLinkUri are added to follow MD034 rule:
+    // https://github.com/DavidAnson/markdownlint/blob/82cf68023f7dbd2948a65c53fc30482432195de4/doc/Rules.md#md034---bare-url-used
+    if (!string.IsNullOrWhiteSpace(helpLinkUri))
+    {
+        helpLinkUri = $"<{helpLinkUri}>";
+    }
+
+    var escapedTitle = descriptor.Title.ToString(CultureInfo.InvariantCulture).Replace("<", "\\<");
+    var line = $"{ruleId} | {helpLinkUri} | {escapedTitle} |";
+    if (validateOnly)
+    {
+        // The validation for RulesMissingDocumentation.md is different than others.
+        // We consider having "extra" entries as valid. This is to prevent CI failures due to rules being documented.
+        // However, we consider "missing" entries as invalid. This is to force updating the file when new rules are added.
+        if (!actualContent.Contains(line))
+        {
+            await Console.Error.WriteLineAsync($"Missing entry in {fileWithPath}").ConfigureAwait(false);
+            await Console.Error.WriteLineAsync(line).ConfigureAwait(false);
+            // The file is missing an entry. Mark it as invalid and break the loop as there is no need to continue validating.
+            return 1;
+        }
+    }
+    else
+    {
+        builder.AppendLine(line);
+    }
+}
+
+if (!validateOnly)
+{
+    File.WriteAllText(fileWithPath, builder.ToString());
+}
+
+
+async Task<bool> checkHelpLinkAsync(string helpLink)
+{
+    try
+    {
+        if (!Uri.TryCreate(helpLink, UriKind.Absolute, out var uri))
+        {
+            return false;
+        }
+
+        var request = new HttpRequestMessage(HttpMethod.Head, uri);
+        using var response = await httpClient.SendAsync(request).ConfigureAwait(false);
+        return response?.StatusCode == HttpStatusCode.OK;
+    }
+    catch (WebException)
+    {
+        return false;
+    }
+}
+
+static SortedList<string, DiagnosticDescriptor> GetAllRulesById(string binDirectory, string configuration)
```

Comment:
Previous method name started with a lower case character. Can we make the casing consistent?